packages/engine/paima-runtime/src/runtime-loops.ts

```diff
@@ -101,10 +102,15 @@ async function runPresync(
           network: network,
           from: height,
           to: upper[network],
-        })),
-        cardanoConfig
+        }))
       );
     });
+
+    if (maybePresyncBlockHeight) {
+      presyncBlockHeight = maybePresyncBlockHeight;
+    } else {
+      break;
+    }
   }
 
   for (const network of Object.keys(networks)) {
```

Comment:
Not related to this PR, but I just noticed the `loopIfStopBlockReached` doesn't work anymore because it's based on `STOP_BLOCKHEIGHT` (single global env) whereas this function checks the env value against the block height for all networks which doesn't really make sense. We should probably open an issue to fix this

Reply:
Right, I haven't really used this feature at all, so I didn't notice.

Reply:
Created an issue: https://github.com/PaimaStudios/paima-engine/issues/324